File: test_sidebar.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

def run_test_sidebar():
    # Initialisation du driver
    driver = webdriver.Chrome()
    driver.maximize_window()

    try:
        # Charger la page principale
        driver.get("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com")
        print("Test du lien 'En savoir plus' (barre latérale)...")

        try:
            # Identifier le lien "En savoir plus" dans la barre latérale
            sidebar_link = WebDriverWait(driver, 15).until(
                EC.element_to_be_clickable((By.XPATH, "//a[contains(text(),'En savoir plus')]"))
            )
            print("✅ Lien 'En savoir plus' trouvé. Tentative de clic...")
            sidebar_link.click()
            
            # Vérifier si un nouvel onglet s'ouvre
            WebDriverWait(driver, 15).until(EC.number_of_windows_to_be(2))
            print("✅ Nouvel onglet détecté. Changement d'onglet...")
            
            # Passer à la nouvelle fenêtre/onglet
            driver.switch_to.window(driver.window_handles[1])
            print(f"URL actuelle dans le nouvel onglet : {driver.current_url}")

            # Vérifier que l'URL cible est atteinte
            assert "faq#security" in driver.current_url, "Redirection incorrecte ou URL inattendue."
            print("✅ Test réussi : la page de la FAQ 'En savoir plus' (barre latérale) s'est ouverte correctement.")

            # Fermer l'onglet de la FAQ
            driver.close()
            driver.switch_to.window(driver.window_handles[0])
            print("✅ Retour à l'onglet principal effectué.")
        except TimeoutException:
            print("❌ Test échoué : impossible de détecter ou cliquer sur le lien 'En savoir plus' (barre latérale).")
    except Exception as e:
        print(f"Erreur inattendue : {e}")
    finally:
        # Fermer le navigateur
        driver.quit()

# Si ce fichier est exécuté directement
if __name__ == "__main__":
    run_test_sidebar()

File: test_sidebar.ts
import { chromium } from 'playwright';

export async function runTestSidebar() {
    const browser = await chromium.launch({ headless: false });
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
        // Charger la page principale
        await page.goto("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com");
        console.log("Test du lien 'En savoir plus' (barre latérale)...");
        await page.waitForTimeout(2000);

        // Identifier le lien "En savoir plus" dans la barre latérale
        const sidebarLink = await page.waitForSelector("//a[contains(text(),'En savoir plus')]");
        console.log("✅ Lien 'En savoir plus' trouvé. Tentative de clic...");
        await sidebarLink.scrollIntoViewIfNeeded(); // S'assure que l'élément est visible
        await page.waitForTimeout(2000);
        await sidebarLink.click();

        // Attendre qu'un nouvel onglet s'ouvre
        const [newPage] = await Promise.all([
            context.waitForEvent('page'),
            page.waitForTimeout(2000)
        ]);

        if (!newPage) {
            throw new Error("❌ L'onglet ne s'est pas ouvert correctement.");
        }
        console.log("✅ Nouvel onglet détecté. Changement d'onglet...");
        await newPage.waitForLoadState();
        console.log(`URL actuelle dans le nouvel onglet : ${newPage.url()}`);
        await page.waitForTimeout(2000);

        // Vérifier que l'URL cible est atteinte
        if (!newPage.url().includes("faq#security")) {
            throw new Error("Redirection incorrecte ou URL inattendue.");
        }
        console.log("✅ Test réussi : la page de la FAQ 'En savoir plus' (barre latérale) s'est ouverte correctement.");

        // Fermer l'onglet de la FAQ et revenir à la fenêtre principale
        await newPage.close();
        console.log("✅ Onglet de la FAQ fermé.");
        await page.bringToFront();
        await page.waitForTimeout(2000);
        console.log("✅ Retour à l'onglet principal effectué.");
    } catch (error) {
        console.error("❌ Test échoué : impossible de détecter ou cliquer sur le lien 'En savoir plus' (barre latérale).", error);
    } finally {
        await browser.close();
    }
}

if (require.main === module) {
    runTestSidebar();
}